File: Java/383. Ransom Note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        // not enough total chars in magazine
        if(ransomNote.length() > magazine.length()){
            return false;
        }
        
        int[] chars = new int[26];
        for(int i=0; i<magazine.length(); i++){
            chars[magazine.charAt(i) - 'a']++;
        }
        
        for(int i=0; i<ransomNote.length(); i++){
            if(chars[ransomNote.charAt(i) - 'a'] <= 0){
                return false;
            }
            chars[ransomNote.charAt(i) - 'a']--;
        }
        
        return true;
    }
}

File: Python3/383. Ransom Note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        # not enough total chars in magazine
        if len(ransomNote) > len(magazine):
            return False
        
        chars = [0] * 26
        a = ord('a')
        
        for c in magazine:
            chars[ord(c) - a] += 1
            
        for c in ransomNote:
            #print("c = %s, chars[c] = %d" %(c, chars[ord(c)-a]))
            # not enough chars for c in magazine
            if chars[ord(c) - a] <= 0:
                return False
            
            chars[ord(c) - a] -= 1
            
        return True
        
        